### PR TITLE
Create Add Course page

### LINKED ISSUE #92: Missing page to add course
**Requirement :**

- Create a Form page so a teacher can add a course.

### PR DESCRIPTION
**Pull request by RanyWh**
**Added a Add Course page to the website that presents the form of the Course**

**Changes :** 
- Add to course/urls a path for the /course/add
- Add a AddCourse class to the views.py with a get and post to handle requests
- Add a form.py to connect between the display and the module
- Add the add_course.html file in templates
- Add the course.add.css to style the add_course.html
- Add the path to the Add Course button in courses.html to send to /course/add

dependencies #83 
fixes #92 

The website :

image

### CODE REVIEW

File: course/templates/course/add_course.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+<html>
+{% load static %}
+    <head>
+        <title>Course</title>
+        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css">
+        <link rel="stylesheet" href="{% static 'css/course.add.css' %}" type="text/css">
+    </head>
+
+
+
```
Comment (shirazyomtov) [inline course/templates/course/add_course.html:13]: Could you delete these empty lines

File: course/templates/course/courses.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+<!DOCTYPE html>
+<html>
+{% load static %}
+    <head>
+        <title>Course</title>
+        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css">
+        <link rel="stylesheet" href="{% static 'css/courses.table.css' %}" type="text/css">
+    </head>
+
+
```
Comment (shirazyomtov) [inline course/templates/course/courses.html:12]: Same as above

File: main/templates/base_home.html [hunk 3]
```diff
@@ -5,7 +5,7 @@
     <title>High.Teach</title>
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <meta charset="utf-8">
-    <link rel="stylesheet" type="text/css" href="{% static 'styles/loggedInPage.css' %}">
+    <link rel="stylesheet" type="text/css" href="{% static 'css/loggedInPage.css' %}">
     <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta2/dist/css/bootstrap.min.css" rel="stylesheet"
```
Comment (shirazyomtov) [inline main/templates/base_home.html:9]: Please change the name of the folder of CSS files to match what appears in the static folder or we can decide on one name for the css folders(There are several PRs with different names for the css folder)

**Review by shirazyomtov — COMMENTED** [review]
Great work, I made some comments

**Review by shirazyomtov — APPROVED** [review]
LGTM

**Review by Davidco94 — DISMISSED** [review]
LGTM

File: course/urls.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+from django.urls import path
+from .views import CourseList, AddCourse
+
+urlpatterns = [
+    path('', CourseList.as_view(), name='show_courses'),
+    path('add', AddCourse.as_view(), name="addCourse"),
```
Comment (natifridman) [inline course/urls.py:6]: Can we try to be consistent in the name convention?

File: tests/course/test_course_pages.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        assert response.status_code == 200
+        assert any(persist_first_student_course.pk == course.pk for course in response.context['courses'])
+
+
+@pytest.mark.django_db
+class TestAddCourseView:
+
+    def test_courses_add_authorized(self, authorized_client):
+        course_response = authorized_client.get('/course/add')
+        assert course_response.status_code == 200
```
Comment (natifridman) [inline tests/course/test_course_pages.py:56]: Maybe we can also check the template in the response?

File: tests/course/test_course_pages.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+    def test_courses_add_unauthorized(self, client):
+        course_response = client.get('/course/add')
+        assert course_response.status_code == 302
```
Comment (natifridman) [inline tests/course/test_course_pages.py:60]: Maybe we can also check the template in the response or the response URL?

File: tests/course/test_course_pages.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+    def test_add_course(self, authorized_client):
+        teacher_course = {"course_name": COURSE_NAME, "description": DESCRIPTION,
+                          "difficulty_level": DIFFICULTY, "category": CATEGORY,
+                          "price": PRICE, "years_of_experience": YEARS_OF_EXP}
+        assert not TeacherCourse.objects.filter(**teacher_course).exists()
+        course_creation_response = authorized_client.post("/course/add", teacher_course)
+        assert course_creation_response.status_code == 302
```
Comment (natifridman) [inline tests/course/test_course_pages.py:68]: Maybe we can also check the template in the response or the response URL?

File: tests/course/test_course_pages.py [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+    def test_show_course_that_user_teaches(self, authorized_client, teacher_course_zero):
+        response = authorized_client.get(reverse('show_courses'))
+        assert response.status_code == 200
+        assert any(teacher_course_zero.pk == course.pk for course in response.context['courses'])
+
+    def test_show_course_that_user_enrolled_in(self, authorized_client, persist_first_student_course):
+        response = authorized_client.get(reverse('show_courses'))
+        assert response.status_code == 200
+        assert any(persist_first_student_course.pk == course.pk for course in response.context['courses'])
```
Comment (Omeramsc) [inline tests/course/test_course_pages.py:48]: for the sake of readability, assign the comprehensions to a variable with a clear name

File: tests/course/test_course_pages.py [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+        assert course_response.status_code == 200
+
+    def test_courses_list_unauthorized(self, client):
+        course_response = client.get('/course/')
+        assert course_response.status_code == 302
+
+    def test_show_course_that_user_teaches(self, authorized_client, teacher_course_zero):
+        response = authorized_client.get(reverse('show_courses'))
+        assert response.status_code == 200
+        assert any(teacher_course_zero.pk == course.pk for course in response.context['courses'])
```
Comment (Omeramsc) [inline tests/course/test_course_pages.py:43]: for the sake of readability, assign the comprehensions to a variable with a clear name

File: tests/course/test_course_pages.py [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+@pytest.mark.django_db
+class TestCourseTableView:
+
+    def test_courses_list_authorized(self, authorized_client):
+        course_response = authorized_client.get('/course/')
+        assert course_response.status_code == 200
+
+    def test_courses_list_unauthorized(self, client):
+        course_response = client.get('/course/')
+        assert course_response.status_code == 302
```
Comment (Omeramsc) [inline tests/course/test_course_pages.py:38]: Maybe we can also check the template in the response or the response URL?

File: tests/course/test_course_pages.py [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+    teacher_course.save()
+    return teacher_course
+
+
+@pytest.mark.django_db
+class TestCourseTableView:
+
+    def test_courses_list_authorized(self, authorized_client):
+        course_response = authorized_client.get('/course/')
+        assert course_response.status_code == 200
```
Comment (Omeramsc) [inline tests/course/test_course_pages.py:34]: Maybe we can also check the template in the response or the response URL?

**Review by Omeramsc — CHANGES_REQUESTED** [review]
left some comments, overall great work. I liked the use of the |= operator.

**Comment by shirazyomtov** [comment]
Please resolve conflicts

**Comment by galitf** [comment]
please fix the conflicts

**Review by natifridman — APPROVED** [review]
LGTM

**Review by shirazyomtov — APPROVED** [review]
LGTM

**Review by idan1356 — APPROVED** [review]
LGTM

**Pull request merged**